openupgradelib/openupgrade_140.py

```diff
@@ -0,0 +1,141 @@
+# Copyright 2023 Tecnativa - Pilar Vargas
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+"""This module provides simple tools for OpenUpgrade migration, specific for
+the 13.0 -> 14.0 migration.
+"""
+
+import logging
+from itertools import product
+
+from psycopg2.extensions import AsIs
+
+from .openupgrade import update_field_multilang
+from .openupgrade_tools import (
+    convert_html_fragment,
+    convert_html_replacement_class_shortcut as _r,
+)
+
+logger = logging.getLogger("OpenUpgrade")
+logger.setLevel(logging.DEBUG)
+
+_PREFIX = ["bg-", "text-", "border-"]
+_CONTEXTS = (
+    ("alpha", 1),
+    ("beta", 2),
+    ("gamma", 3),
+    ("delta", 4),
+    ("epsilon", 5),
+    ("zeta", 6),
+    ("eta", 7),
+)
+
+# These replacements are specific for Odoo v13 to v14
+_ODOO14_REPLACEMENTS = (
+    *(
+        _r(f"{prefix}{v13}", f"{prefix}o-color-{v14}")
+        for prefix, (v13, v14) in product(_PREFIX, _CONTEXTS)
+    ),
+    *(_r(f"btn-{v13}", f"bg-o-color-{v14}") for (v13, v14) in _CONTEXTS),
+    *(_r(f"o-mm-def-color-{v13}", f"o-mm-def-color-{v14}") for (v13, v14) in _CONTEXTS),
```

Comment:
AFAIK, these are SCSS variables:

https://github.com/odoo/odoo/commit/7736cde22c3bce06305045a818a2d70892cf25ef#diff-0c5e52d5ad8371cd84ae9c059ce38825062399653c4892d22d6c8f9a7e2267f3R6

so no need to rename it. It would make sense if touching custom CSS files, but that's also a weird thing, and not sure right now how are they stored. I think they are put as attachments. cc @chienandalu

Reply:
What about this?

Reply:
No, that line should go away as it isn't something being used in the qweb templates. CSS rules should be adapted in each theme/module migration.

Reply:
@pilarvargas-tecnativa this comment is pending.

Reply:
sorry, corrected